source/dub/recipe/packagerecipe.d

```diff
@@ -171,50 +213,105 @@ struct ConfigurationInfo {
 	}
 }
 
+/// Actual representation of a dependency as specified by the user
+private struct DependencyObject
+{
+    bool optional;
+    @Optional @Name("version")
+    string version_;
+    @Optional string path;
+}
```

Comment:
this struct is missing `repository` and `default`

also make `optional` and `version_` `@Optional`, there are no required fields in dependency, these are all valid:
```
{"path":"mydependency"}
{"repository": "git+https://github.com/dlang-community/gitcompatibledubpackage.git","version": "ccb31bf6a655437176ec02e04c2305a8c7c90d67"}
"0.1.0"
{"version":"0.1.0","optional":true,"default":false}
```

also:
> Since v1.24.0, [build settings](https://dub.pm/package-format-json.html#build-settings) fields can be used to be applied to dependencies.
> 
> Note that as of v1.24.0 only dflags is taken into account.

Reply:
Right, will need to fix this.